### PR TITLE
Adding functor unit tests 

### PR DESCRIPTION
**Pull request by rpathani**
Please review the unit tests written for funtor

### CODE REVIEW

**Comment by AlexVlx** [comment]
Why does this exist? hipLaunchParam is deprecated and has been deprecated for a while, why are we messing with it and creating overhead?

File: tests/src/kernel/hipLaunchParmFunctor.cpp [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+/* HIT_START
+ * BUILD: %t %s ../test_common.cpp
+ * RUN: %t
+ * HIT_END
+ */
+
+#include <iostream>
+#include "hip/hip_runtime.h"
+#include "../test_common.h"
+#include "hip/hip_runtime_api.h"
```
Comment (gargrahul) [inline tests/src/kernel/hipLaunchParmFunctor.cpp]: Is this needed?

File: tests/src/kernel/hipLaunchParmFunctor.cpp [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+/* HIT_START
+ * BUILD: %t %s ../test_common.cpp
```
Comment (gargrahul) [inline tests/src/kernel/hipLaunchParmFunctor.cpp]: Does this run fine on HIP/NVCC path?

File: tests/src/kernel/hipLaunchParmFunctor.cpp [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+           hipMemcpyDeviceToHost));
+  for (int k = 0; k < BLOCK_DIM_SIZE; ++k)
+    HIPASSERT(hostResults[k] == true);
+  HIPCHECK(hipHostFree(hostResults));
+  HIPCHECK(hipFree(deviceResults));
+}
+
+
+// Doubler calculator
+class DoublerCaluclator {
```
Comment (gargrahul) [inline tests/src/kernel/hipLaunchParmFunctor.cpp]: Calculator is misspelled.

File: tests/src/kernel/hipLaunchParmFunctor.cpp (hunk 1 above)
Comment (rpathani) [reply tests/src/kernel/hipLaunchParmFunctor.cpp]: Only  #include "../test_common.h" is required , Rest can be removed

File: tests/src/kernel/hipLaunchParmFunctor.cpp (hunk 2 above)
Comment (rpathani) [reply tests/src/kernel/hipLaunchParmFunctor.cpp]: @gargrahul Not checked on HIP/NVCC  path yet. Verified on HIP/HCC path. Tests work fine on HIP/HCC path
On HIP-CLANG path the tests are passing with -O3 option but without -O3 option it is not able to compile functor

File: tests/src/kernel/hipLaunchParmFunctor.cpp (hunk 1 above)
Comment (rpathani) [reply tests/src/kernel/hipLaunchParmFunctor.cpp]: Done

File: tests/src/kernel/hipLaunchParmFunctor.cpp (hunk 2 above)
Comment (rpathani) [reply tests/src/kernel/hipLaunchParmFunctor.cpp]: Verified the tests on HIP/NVCC and HIP/HCC and clang path.

File: tests/src/kernel/hipLaunchParmFunctor.cpp (hunk 3 above)
Comment (rpathani) [reply tests/src/kernel/hipLaunchParmFunctor.cpp]: Thanks ,corrected.

**Pull request merged**